src/containers/Home/Restaurant.js

```diff
@@ -0,0 +1,229 @@
+const key = Config.GOOGLE_MAPS_API_KEY;
+
+class Restaurant extends React.Component<Props, State> {
+  static props = {
+    visibility: false,
+    photo: [],
+    place: null,
+    listTitle: '',
+  };
+
+  state = {
+    photos: [],
+    listFound: false,
+  };
+
+  componentWillReceiveProps(nextProps) {
+    if (this.props.photo !== nextProps.photo) {
+      if (nextProps.photo.length > 0) {
+        // fetch 4 photos from random starting index point
+        if (nextProps.photo.length > 4) {
+          let startPoint = nextProps.photo.length;
+          while (startPoint + 3 > nextProps.photo.length - 1) {
+            startPoint = Math.floor(Math.random() * (nextProps.photo.length));
+          }
+          this.fetchPhoto(nextProps.photo[startPoint]);
+          this.fetchPhoto(nextProps.photo[startPoint + 1]);
+          this.fetchPhoto(nextProps.photo[startPoint + 2]);
+          this.fetchPhoto(nextProps.photo[startPoint + 3]);
+        } else {
+          this.fetchPhoto(nextProps.photo[0]);
+          this.fetchPhoto(nextProps.photo[1]);
+          this.fetchPhoto(nextProps.photo[2]);
+          this.fetchPhoto(nextProps.photo[3]);
+        }
+      }
+    }
+  }
+
+  componentDidUpdate() {
+    if (this.props.listTitle !== '') {
+      let found = false;
+      // check if list existed
+      for (let i = 0; i < this.props.favouriteList.length; i += 1) {
+        const title = this.props.favouriteList.map((item) => {
+          const temp = item.title;
+          return temp;
+        });
+        if (title[i] === this.props.listTitle) {
+          found = true;
+          break;
+        }
+      }
+      if (found) {
+        this.listFound();
+      }
+    }
+  }
+
+  // set listFound to true when list is found
+  listFound() {
+    if (!this.state.listFound) {
+      this.setState({
+        listFound: true,
+      });
+    }
+  }
+
+  // fetch photo url using photo reference id
+  fetchPhoto(referenceId): void {
+    const params = {
+      photoreference: referenceId,
+      maxheight: 150,
+      maxweight: 150,
+      key,
+    };
+
+    fetch(
+      `${PLACES_PHOTO_API}${querystring.stringify(params)}`,
+      {
+        method: 'HEAD',
+      },
+    )
+      .then((response) => {
+        if (response) {
+          const tempPhotos = this.state.photos;
+          // prevent dummy photos
+          if (response.url.search('googleapis') === -1) {
+            tempPhotos.push(response.url);
+            // console.log(tempPhotos);
+            this.setState({
+              photos: tempPhotos,
+            });
+          }
+        }
+      })
+      .catch(() => {
+        // Retry 5s later, inhibiting errors
+        setTimeout(
+          this.fetchPhoto(referenceId),
+          5000,
+        );
+      });
+  }
+
+  handleBackButton() {
+    if (this.props.visibility) {
+      this.props.doSetRes(false);
+      this.props.doDumpPhoto();
+      this.state.photos = [];
```

Comment:
Do not mutate the state value but use this.setState() to vary the state

Reply:
sorry, my mistake